docs/architecture/adr-042-group-module.md

```diff
@@ -0,0 +1,375 @@
+# ADR 042: Group Module
+
+## Changelog
+
+- 2020/04/09: Initial Draft
+
+## Status
+
+Draft
+
+## Abstract
+
+This ADR defines the `x/group` module which allows the creation and management of on-chain multi-signature accounts and enables voting for message execution based on configurable decision policies.
+
+## Context
+
+The current multi-signature mechanism of the Cosmos SDK has certain limitations:
+- Key rotation is not possible, although this can be solved with [account rekeying](adr-034-account-rekeying.md).
+- Thresholds can't be changed.
+- UX is not straightforward for non-technical users ([#5661](https://github.com/cosmos/cosmos-sdk/issues/5661)).
+- It requires `legacy_amino` sign mode ([#8141](https://github.com/cosmos/cosmos-sdk/issues/8141)).
+
+While the group module is not meant to be a total replacement for the current multi-signature accounts, it provides a solution to the limitations described above, with a more flexible key management system where keys can be added, updated or removed, as well as configurable thresholds.
+It's meant to be used with other key management modules such as [`x/feegrant`](./adr-029-fee-grant-module.md) ans [`x/authz`](adr-030-authz-module.md) to simplify key management for individuals and organizations.
+
+The current implementation of the group module can be found in https://github.com/regen-network/regen-ledger/tree/master/proto/regen/group/v1alpha1 and https://github.com/regen-network/regen-ledger/tree/master/x/group.
+
+## Decision
+
+We propose merging the `x/group` module with its supporting [ORM/Table Store package](https://github.com/regen-network/regen-ledger/tree/master/orm) ([#7098](https://github.com/cosmos/cosmos-sdk/issues/7098)) into the Cosmos SDK and continuing development here. There will be a dedicated ADR for the ORM package.
+
+### Group
+
+A group is an aggregation of accounts with associated weights. It is not
+an account and doesn't have a balance. It doesn't in and of itself have any
+sort of voting or decision weight. It has an `admin` account which can manage members in the group.
```

Comment:
Oh, I have an answer - this is explained in the `GroupAccount` section. For having an easier read, could you make a paragraph before the `Group` section to roughly talk about Group and GroupAccount (what's the use-case for each and what's the relationship between them)?

Reply:
> > It doesn't in and of itself have any sort of voting or decision weight.
> 
> Not sure what do you mean with that? Group account can be used in another group account (and vote), right?
> 
> Why we don't allow keepeing a balance for a group account? It has an address and it can use `x/bank` , right?

We're talking about groups here, not group accounts.

Reply:
> Oh, I have an answer - this is explained in the `GroupAccount` section. For having an easier read, could you make a paragraph before the `Group` section to roughly talk about Group and GroupAccount (what's the use-case for each and what's the relationship between them)?

ok I'll try to make this clearer